wgpu/src/backend/custom.rs

```diff
@@ -0,0 +1,88 @@
+//! Provides wrappers custom backend implementations
+
+#![allow(ambiguous_wide_pointer_comparisons)]
+
+pub use crate::dispatch::*;
+
+use std::sync::Arc;
+
+macro_rules! dyn_type {
+    // cloning of arc forbidden
+    (pub mut struct $name:ident(dyn $interface:tt)) => {
+        #[derive(Debug)]
+        pub(crate) struct $name(Arc<dyn $interface>);
```

Comment:
Any reason this couldn't be `Box`?

Reply:
It's to obtain `Eq`,`Hash`,`Cmp` implementations. We cannot use T implementations because traits are not object safe. Will write in comment.